library/src/G4CMPInterValleyScattering.cc

```diff
@@ -102,42 +107,222 @@ G4double G4CMPInterValleyScattering::GetMeanFreePath(const G4Track& track,
 
 // Perform scattering action
 
-G4VParticleChange* 
-G4CMPInterValleyScattering::PostStepDoIt(const G4Track& aTrack, 
-					 const G4Step& aStep) {
-  InitializeParticleChange(GetValleyIndex(aTrack), aTrack);
-  G4StepPoint* postStepPoint = aStep.GetPostStepPoint();
-  
-  if (verboseLevel > 1) {
-    G4cout << GetProcessName() << "::PostStepDoIt: Step limited by "
-	   << postStepPoint->GetProcessDefinedStep()->GetProcessName()
-	   << G4endl;
-  }
-  
-  // Don't do anything at a volume boundary
-  if (postStepPoint->GetStepStatus()==fGeomBoundary) {
-    return G4VDiscreteProcess::PostStepDoIt(aTrack, aStep);
-  }
-  
-  // Get track's energy in current valley
-  G4ThreeVector p = GetLocalMomentum(aTrack);
-  G4int valley = GetValleyIndex(aTrack);
-  p = theLattice->MapPtoK(valley, p); // p is actually k now
-  p = theLattice->RotateToValley(valley, p);
-  
-  // picking a new valley at random if IV-scattering process was triggered
-  valley = ChangeValley(valley);
-  G4CMP::GetTrackInfo<G4CMPDriftTrackInfo>(aTrack)->SetValleyIndex(valley);
-
-  p = theLattice->RotateFromValley(valley, p);
-  p = theLattice->MapKtoP(valley, p); // p is p again
-  RotateToGlobalDirection(p);
-  
-  // Adjust track kinematics for new valley
-  FillParticleChange(valley, p);
-  
-  ClearNumberOfInteractionLengthLeft();    
-  return &aParticleChange;
+G4VParticleChange*
+G4CMPInterValleyScattering::PostStepDoIt(const G4Track& aTrack,
+                                         const G4Step& aStep) {
+    aParticleChange.Initialize(aTrack);
```

Comment:
This whole function is indented too far.  We use two-space indentation for each scoping level.  Can you adjust your editor to match that?